homeassistant/components/here_weather/config_flow.py

```diff
@@ -0,0 +1,244 @@
+                    title=user_input[CONF_NAME], data=user_input
+                )
+            except AlreadyConfigured:
+                return self.async_abort(reason="already_configured")
+            except herepy.InvalidRequestError:
+                errors["base"] = "invalid_request"
+            except herepy.UnauthorizedError:
+                errors["base"] = "unauthorized"
+        return self.async_show_form(
+            step_id="coordinates",
+            data_schema=get_coordinate_schema(self.hass),
+            errors=errors,
+        )
+
+    async def async_step_zip_code(self, user_input=None):
+        """Handle set up by zip_code."""
+        errors = {}
+        if user_input is not None:
+            try:
+                await async_validate_zip_code_input(self.hass, user_input)
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
+                )
+            except AlreadyConfigured:
+                return self.async_abort(reason="already_configured")
+            except herepy.InvalidRequestError:
+                errors["base"] = "invalid_request"
+            except herepy.UnauthorizedError:
+                errors["base"] = "unauthorized"
+        return self.async_show_form(
+            step_id="zip_code",
+            data_schema=get_zip_code_schema(self.hass),
+            errors=errors,
+        )
+
+    async def async_step_location_name(self, user_input=None):
+        """Handle set up by location_name."""
+        errors = {}
+        if user_input is not None:
+            try:
+                await async_validate_location_name_input(self.hass, user_input)
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
+                )
+            except AlreadyConfigured:
+                return self.async_abort(reason="already_configured")
+            except herepy.InvalidRequestError:
+                errors["base"] = "invalid_request"
+            except herepy.UnauthorizedError:
+                errors["base"] = "unauthorized"
+        return self.async_show_form(
+            step_id="location_name",
+            data_schema=get_location_name_schema(self.hass),
+            errors=errors,
+        )
+
+
+class HereWeatherOptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle here_weather options."""
+
+    def __init__(self, config_entry):
+        """Initialize here_weather options flow."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the here_weather options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        options = {
+            vol.Optional(
+                CONF_SCAN_INTERVAL,
+                default=self.config_entry.options.get(
+                    CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
```

Comment:
Why is scan interval useful? Can we set a sane default instead?

Reply:
If the users want to save on their free api call volume they can increase the interval. If they want to have updates every 5 minutes this can be done. Why should we take that option away?

Reply:
We try to avoid config options where the user can configure themselves into a bad situation, like being rate limited. If you think this options is essential, we can keep it.

Reply:
The user can always add more than one instance of this integration if more than one location should be tracked. We cannot control all possible combinations which would lead to a rate limit

Reply:
So how other integrations solve this is by adjusting the rate limit based on the number of active config entries.

If the user wants a different schedule/schema, we have resolve that generic, see: <https://www.home-assistant.io/blog/2021/06/02/release-20216/#disable-polling-updates-on-any-integration>

For that case, we don't have to implement a scan interval.

I agree with Martin, we should have a sane default (e.g., by calculating based on the number of entries) and not expose this option.

Reply:
Fixed in https://github.com/home-assistant/core/pull/28910/commits/388ee89d5a35380fd94dd09d6963302e12f7ea8d